tests/ga/test_remoteaccess_handler.py

```diff
@@ -58,478 +82,431 @@ def setUp(self):
-        users = get_user_dictionary(rah.os_util.get_users())
-        self.assertTrue(tstuser in users, "{0} missing from users".format(tstuser))
-        actual_user = users[tstuser]
-        expected_expiration = (expiration_date + timedelta(days=1)).strftime("%Y-%m-%d")
-        self.assertEqual(actual_user[7], expected_expiration)
-        self.assertEqual(actual_user[4], "JIT_Account")
-        self.assertEqual(0, len(error_messages))
-        self.assertEqual(1, len(info_messages))
-        self.assertEqual(info_messages[0], "User '{0}' added successfully with expiration in {1}"
-                         .format(tstuser, expected_expiration))
-
-    @patch('azurelinuxagent.common.logger.Logger.info', side_effect=log_info)
-    @patch('azurelinuxagent.common.logger.Logger.error', side_effect=log_error)
+    def test_add_user(self, *_):
+        with patch("azurelinuxagent.ga.remoteaccess.get_osutil", return_value=MockOSUtil()):
+            rah = RemoteAccessHandler(Mock())
+            tstpassword = "]aPPEv}uNg1FPnl?"
+            tstuser = "foobar"
+            expiration_date = datetime.utcnow() + timedelta(days=1)
+            pwd = tstpassword
+            rah._add_user(tstuser, pwd, expiration_date)
+            users = get_user_dictionary(rah._os_util.get_users())
+            self.assertTrue(tstuser in users, "{0} missing from users".format(tstuser))
+            actual_user = users[tstuser]
+            expected_expiration = (expiration_date + timedelta(days=1)).strftime("%Y-%m-%d")
+            self.assertEqual(actual_user[7], expected_expiration)
+            self.assertEqual(actual_user[4], "JIT_Account")
+
     @patch('azurelinuxagent.common.utils.cryptutil.CryptUtil.decrypt_secret', return_value="]aPPEv}uNg1FPnl?")
-    def test_add_user_bad_creation_data(self, _1, _2, _3):
-        rah = RemoteAccessHandler(Mock())
-        rah.os_util = MockOSUtil()
-        tstpassword = "]aPPEv}uNg1FPnl?"
-        tstuser = ""
-        expiration = datetime.utcnow() + timedelta(days=1)
-        pwd = tstpassword
-        error = "Error adding user {0}. test exception for bad username".format(tstuser)
-        self.assertRaisesRegex(RemoteAccessError, error, rah.add_user, tstuser, pwd, expiration)
-        self.assertEqual(0, len(rah.os_util.get_users()))
-        self.assertEqual(0, len(error_messages))
-        self.assertEqual(0, len(info_messages))
-
-    @patch('azurelinuxagent.common.logger.Logger.info', side_effect=log_info)
-    @patch('azurelinuxagent.common.logger.Logger.error', side_effect=log_error)
+    def test_add_user_bad_creation_data(self, *_):
+        with patch("azurelinuxagent.ga.remoteaccess.get_osutil", return_value=MockOSUtil()):
+            rah = RemoteAccessHandler(Mock())
+            tstpassword = "]aPPEv}uNg1FPnl?"
+            tstuser = ""
+            expiration = datetime.utcnow() + timedelta(days=1)
+            pwd = tstpassword
+            error = "test exception for bad username".format(tstuser)
```

Comment:
This format (line 113) doesn't seem to do anything. Is the intent here to include the test username in the error string? or is the format call extraneous?

Reply:
thanks; i forgot to remove the call to format -- there may be some other instances below; i'll check all of them

Reply:
yes, i did find a few more. all fixed now